lib/atlas/dataset/derived_dataset.rb

```diff
@@ -0,0 +1,63 @@
+module Atlas
+  class Dataset::DerivedDataset < Dataset
```

Comment:
How would you guys feel about naming these two new classes simply "Derived"  and "Full"? "Dataset" already appears in the module tree, and it would shorten the .ad file names ("datasets/nl/nl.full.ad" instead of "datasets/nl/nl.full_dataset.ad").

It would also be more consistent [with Node subclasses](https://github.com/quintel/atlas/tree/1a67953fc974177bad9eb194750d4677b4b99880/lib/atlas/node) (although I think consistency is not super-important in this case).